perun-core/src/test/java/cz/metacentrum/perun/core/entry/ConsentsManagerEntryIntegrationTest.java

```diff
@@ -32,6 +45,185 @@ public void setUp() throws Exception {
 		consentsManagerEntry = perun.getConsentsManager();
 	}
 
+
+	@Test
+	public void createConsent() throws Exception {
+		System.out.println(CLASS_NAME + "createConsent");
+
+		User user = setUpUser("John", "Doe");
+
+		Facility facility = setUpFacility();
+
+		Consent consent = new Consent(-1, user.getId(), perun.getConsentsManager().getConsentHubByName(sess, facility.getName()), new ArrayList<>());
+		consent = perun.getConsentsManagerBl().createConsent(sess, consent);
+
+		assertEquals(consent, consentsManagerEntry.getConsentById(sess, consent.getId()));
+	}
+
+	@Test (expected = ConsentNotExistsException.class)
+	public void createConsentDeleteExistingUnsigned() throws Exception {
+		System.out.println(CLASS_NAME + "createConsentDeleteExistingUnsigned");
+
+		User user = setUpUser("John", "Doe");
+
+		Facility facility = setUpFacility();
+
+		Consent consentOld = new Consent(-1, user.getId(), perun.getConsentsManager().getConsentHubByName(sess, facility.getName()), new ArrayList<>());
+		consentOld = perun.getConsentsManagerBl().createConsent(sess, consentOld);
+
+		Consent consentNew = new Consent(-11, user.getId(), perun.getConsentsManager().getConsentHubByName(sess, facility.getName()), new ArrayList<>());
+		consentNew = perun.getConsentsManagerBl().createConsent(sess, consentNew);
+
+		perun.getConsentsManagerBl().checkConsentExists(sess, consentOld);
+
+	}
+
+	@Test (expected = ConsentNotExistsException.class)
+	public void deleteConsent() throws Exception {
+		System.out.println(CLASS_NAME + "deleteConsent");
+
+		User user = setUpUser("John", "Doe");
+
+		Facility facility = setUpFacility();
+
+		Consent consent = new Consent(-1, user.getId(), perun.getConsentsManager().getConsentHubByName(sess, facility.getName()), new ArrayList<>());
+		perun.getConsentsManagerBl().deleteConsent(sess, consent);
```

Comment:
I think this test is the case I was talking about above. I believe the exception `ConsentNotExistsException` would be thrown already here. We should also probably have two separate tests - one for deleting existing consent and one for deleting not-existing one.